contracts/swear.sol

```diff
@@ -0,0 +1,144 @@
+pragma solidity ^0.4.0;
+
+import "./abstracts/swearabstract.sol";
+import "./abstracts/trialrulesabstract.sol";
+import "./abstracts/token.sol";
+import "./sampletoken.sol";
+
+
+contract Swear is SwearAbstract {
+
+    mapping(address=>Deposit) public deposits;
+    mapping(address => bool) public players;
+
+    uint  public playerCount;
+    SampleToken public token;
+    TrialRulesAbstract public trialRules;
+    address swindle;
+    mapping(address => uint256)  OpenCases;
+
+    struct Deposit {
+        bool inDepositPeriod;
+        uint vestingPeriod;
+        uint depositedAmount;
+    }
+
+    function Swear(address _trialRules,address _token) {
+        token = SampleToken(_token);
+        trialRules = TrialRulesAbstract(_trialRules);
+    }
+
+    function incrementOpenCases(address _address) {
+        require(msg.sender == swindle);
+        OpenCases[_address]++;
+    }
+
+    function decrementOpenCases(address _address) {
+        require(msg.sender == swindle);
+        OpenCases[_address]--;
+    }
+
+    /// @notice register - register a player to the game
+    ///
+    /// The function will throw if the player is already register or there is not
+    /// enough deposit in the contract to ensure the player could be compensated for the
+    /// case of a valid case.
+    /// @param _player  - the player address
+    /// @return bool registered - true for success registration.
+    function register(address _player) onlyOwner public returns (bool) {
```

Comment:
wonder if we want to restrict register to  only owner. 
The typical use case is that these are multiple provider services where anyone can registter and pay a fixed amount.

In fact we could consider a super  generic swear contract which allows you to register/commit to any game. 
This is because registration, deposit,  compensation is independent of rules.